python/tests/stress/arcticdb/version_store/test_stress_finalize_staged_data.py

```diff
@@ -85,8 +85,11 @@ def test_finalize_monotonic_unique_chunks(basic_arctic_library):
     print(f"Writing to symbol initially {num_rows_initially} rows")
     df = cachedDF.generate_dataframe_timestamp_indexed(num_rows_initially, total_number_rows, cachedDF.TIME_UNIT)
 
+    iterations = [500, 1000, 1500, 2000] 
+    if ("amazonaws" in lib.arctic_instance_desc.lower()):
```

Comment:
instead of putting ifs in the test making it less reliable to the environments, i would suggest to create fixtures and pass iterations to tests. like this you can create a separate test for amazonaws which will be much better visible for the failures and in test results

Reply:
I am not sure that this is possible with current design of fixtures ... Currently I use fixture called 'basic_arctic_library'. It automatically creates 6 version of the test. 2 of which are S3 AWS with 2 different types of encodings.

We do not have actually a way to combine environments or select only environments we want from a fixtere that has been predefined with such. Or At least I am not aware how this can happen. 

I will see what I can do with current set of fixtures we have. Otherwise combining couple of this and that in a new fixture results in huge pile of fixtures which again is going to be very complex to maintain

Reply:
You can create fixture that returns library and the sizes in tuple or dictionary, which can be fed into this test. Fixtures can also create multiple params in tuple.